crates/bevy_ui/src/widget/slider.rs

```diff
@@ -0,0 +1,171 @@
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_ecs::query::With;
+use bevy_ecs::reflect::ReflectComponent;
+use bevy_ecs::schedule::SystemLabel;
+use bevy_ecs::system::{Query, Res};
+use bevy_hierarchy::Children;
+use bevy_input::prelude::MouseButton;
+use bevy_input::touch::Touches;
+use bevy_input::Input;
+use bevy_reflect::std_traits::ReflectDefault;
+use bevy_reflect::Reflect;
+use thiserror::Error;
+
+use crate::{Interaction, Node, RelativeCursorPosition, Style, Val};
+
+/// A component describing the slider-specific value, such as max and min values and step
+#[derive(Component, Debug, Clone, Copy, Reflect)]
+#[reflect(Component, Default)]
+pub struct Slider {
+    min: f32,
+    max: f32,
+    step: f32,
+    value: f32,
+}
+
+impl Default for Slider {
+    fn default() -> Self {
+        Self {
+            min: 0.,
+            max: 100.,
+            step: 0.,
+            value: 0.,
+        }
+    }
+}
+
+impl Slider {
+    /// Creates a new `Slider` with `min` and `max` values
+    /// `Min` and `max` don't affect the physical size of the slider, they're only used for calculating the value of the slider
+    pub fn new(min: f32, max: f32) -> Self {
+        Self {
+            min,
+            max,
+            step: 0.,
+            value: min,
+        }
+    }
+
+    /// Consumes self, returning a new [`Slider`] with a given value
+    pub fn with_value(self, value: f32) -> Self {
+        Self { value, ..self }
+    }
+
+    /// Consumes self, returning a new [`Slider`] with a given step
+    pub fn with_step(self, step: f32) -> Self {
+        Self { step, ..self }
+    }
+
+    /// Sets the slider value, returning error if the given value is out of the slider range
+    pub fn set_value(&mut self, value: f32) -> Result<(), SliderValueError> {
+        // Round the value up to self.step (we have to consider that self.min can be a fraction)
+        let value = if self.step != 0. {
+            ((value - self.min) / self.step).round() * self.step + self.min
+        } else {
+            value
+        };
+
+        if (self.min..=self.max).contains(&value) {
+            self.value = value;
+            return Ok(());
+        }
+
+        Err(SliderValueError::ValueOutOfSliderRange)
+    }
+
+    /// Retrieves the slider value
+    pub fn get_value(&self) -> f32 {
```

Comment:
Most bevy api's use get_* for fallible getters and don't use any prefix when it can't fail. I'd suggest changing all getter to just the name without prefix.